### ISSUE TITLE
[rpc] interface conversion: interface {} in nil, not EventDataTx

### ISSUE THREAD

**Issue by melekes** [tendermint/tendermint#2408]
Reported via Riot.

<img width="1289" alt="rpc-error" src="https://user-images.githubusercontent.com/1282182/45587834-68023c80-b91c-11e8-97fc-b63805d63701.png">

**Comment by bradyjoestar** [comment]
I will try to  solve this bug ,but how to reproduce it?
Thanks.

**Comment by melekes** [comment]
I asked, but did not receive a response.

**Comment by melekes** [comment]
Log: [logs.zip](https://github.com/tendermint/tendermint/files/2388185/logs.zip)

**Comment by bradyjoestar** [comment]
sorry for being late.
I think this error was caused by broadcast_tx_commit through rpc. :smile: 

This error was caused by the following steps:
1.First a new block come out and include many transactions.
2.Tendermint call ABCIAPP and deliverTx to it.
3.At this time a suddenly interrupt (i.e `CTRL+C`) happened.
4.Deliver tx failed.
5.but rpc-server and other service not failed, it continue to call `broadcast_tx_commit`(I searched in the tendermint and not found where it is),and pass a interface{}  not types.EventDataTx to rpc-server.
6.This error happened.
 

I will try to do something to prevent it. :smile:
